src/main/java/com/example/hop_oasis/service/data/UserAuthenticated.java

```diff
@@ -0,0 +1,25 @@
+package com.example.hop_oasis.service.data;
+
+import com.example.hop_oasis.model.User;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Service;
+
+@Service
+public class UserAuthenticated {
+
+    public User getAuthenticatedUser() {
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+        if (authentication != null && authentication.isAuthenticated() &&
+                authentication.getPrincipal() instanceof User) {
+            User currentUser = (User) authentication.getPrincipal();
+
+            if (currentUser != null) {
+                return currentUser;
+            } else {
+                throw new RuntimeException("User not found");
+            }
+        }
+        return null;
```

Comment:
it doesn't make a sense. You should either throw an exception or return Optional. But above you already throwing an exception. 
Your one method can return null, throw an exception or return an object. That's way too much)